zomes/elements/code/src/lib.rs

```diff
@@ -49,15 +54,71 @@ fn element_entry () -> ValidatingEntryType {
     }
 }
 
-fn handle_contribute_element(element: Element) -> ZomeApiResult<Address> {
-    hdk::debug(format!("handle_contribute_element({:?})", element))?;
-    let new_entry = Entry::App("Element".into(), element.into());
-    let address = hdk::commit_entry(&new_entry)?;
+fn element_index_entry () -> ValidatingEntryType {
+    entry! {
+        name: "ElementIndex",
+        description: "Entries that link to all elements of a type",
+        sharing: Sharing::Public,
+
+        validation_package: || {
+            hdk::ValidationPackageDefinition::Entry
+        },
+
+        validation: |_validation_data: hdk::EntryValidationData<ElementIndex>| {
+            Ok(())
+        },
+
+        links: [
+            to! {
+                "Element",
+                link_type: "Index",
+
+                validation_package : || {
+                    hdk::ValidationPackageDefinition::Entry
+                },
+                validation: |validation_data: hdk::LinkValidationData| {
+                    if let hdk::LinkValidationData::LinkAdd{
+                            link: LinkData{
+                                link: link_,
+                                action_kind: _
+                            }, 
+                            validation_data: _
+                        } = validation_data {
+                        let base = handle_get_element_index(link_.base().to_owned())?;
+                        let target = handle_get_element(link_.target().to_owned())?;
+                        return valid_base_and_target(&base, &target);
+                    } else {
+                       return Err("Cannot remove links at this time".to_string());
```

Comment:
Out of context this is confusing, it's not immediately clear that the this else branch is when the if let is for a LinkRemove.

Maybe a line comment would be helpful